common/flatpak-dir.c

```diff
@@ -12090,6 +12114,15 @@ _flatpak_dir_get_remote_state (FlatpakDir   *self,
       if (var_metadata_lookup (meta, "xa.default-token-type", NULL, &res) &&
           var_variant_is_type (res, G_VARIANT_TYPE_INT32))
         state->default_token_type = GINT32_FROM_LE (var_variant_get_int32 (res));
+
+      if (var_metadata_lookup (meta, "xa.debuginfod-urls", NULL, &res) &&
+          var_variant_is_type (res, G_VARIANT_TYPE_STRING_ARRAY))
+          {
+            g_autoptr(GVariant) variant = g_variant_ref_sink (var_variant_peek_as_variant (res));
+            g_autoptr(GVariant) child = g_variant_get_variant (variant);
+            g_autofree const char **debuginfod_urls = g_variant_get_strv (child, NULL);
```

Comment:
I wrote this by experimentation - maybe there's a cleaner way to do this?

Reply:
You should be able to do 
g_autofree char **debuginfo_urls = var_arrayofstring_to_strv (var_arrayofstring_from_variant (res));